expression_filter.cpp

```diff
@@ -152,9 +153,10 @@ int				varCount = 0;
 
 				// We have added a new variable so must re-parse the expression
 				expression.register_symbol_table(symbolTable);
-				if (!parser.compile(m_expression.c_str(), expression))
+				if (parser.compile(m_expression.c_str(), expression) == false && m_report)
 				{
 					Logger::getLogger()->error("Expression compilation failed: %s", parser.error().c_str());
```

Comment:
Is it really an `error`? I believe we should log as a `warning`.
Also it would be better if we add asset_code or datapoint name in the log message for which the expression compilation failed.

Reply:
I would say it is an error as it is effectively saying there is an undefined variable in the expression.